test/index.js

```diff
@@ -111,15 +111,19 @@ test('exhaustive vs non-exhaustive expansion', function (t) {
 
 test('raw curly braces for custom slot types', function (t) {
   var dictionary = {};
-  var slots = {"Artist": "CUSTOM_TYPE"};
-  var template = "{my|your} {favorite|least favorite} fruit is {-|Fruit}";
+  var slots = {"Artist": "CUSTOM_TYPE","ROOM_NAME": "AMAZON.Room"};
+  var template = "{my|your} {favorite|least favorite} fruit is {-|Fruit} {in +ROOM_NAME+|}";
 
   var result = utterances(template, slots, dictionary);
   t.deepEqual(result, [
-    "my favorite fruit is {Fruit}",
-    "your favorite fruit is {Fruit}",
-    "my least favorite fruit is {Fruit}",
-    "your least favorite fruit is {Fruit}"
+    "my favorite fruit is {Fruit} in {ROOM_NAME}",
+    "your favorite fruit is {Fruit} in {ROOM_NAME}",
+	"my least favorite fruit is {Fruit} in {ROOM_NAME}",
+    "your least favorite fruit is {Fruit} in {ROOM_NAME}",
+	"my favorite fruit is {Fruit} ",
```

Comment:
No doubt. Here's what I would do:

1. Update the code to trim the whitespace (or not add it in the first place)
1. Add a *new* test for the optional terms which doesn't include the trailing white space
1. Run all the tests with your new code. They should all pass
1. Report back if you have issues